sherpa-onnx/c-api/c-api.cc

```diff
@@ -410,6 +411,184 @@ void DestroyOfflineRecognizerResult(
+  spotter_config.feat_config.feature_dim =
+      SHERPA_ONNX_OR(config->feat_config.feature_dim, 80);
+
+  spotter_config.model_config.transducer.encoder =
+      SHERPA_ONNX_OR(config->model_config.transducer.encoder, "");
+  spotter_config.model_config.transducer.decoder =
+      SHERPA_ONNX_OR(config->model_config.transducer.decoder, "");
+  spotter_config.model_config.transducer.joiner =
+      SHERPA_ONNX_OR(config->model_config.transducer.joiner, "");
+
+  spotter_config.model_config.paraformer.encoder =
+      SHERPA_ONNX_OR(config->model_config.paraformer.encoder, "");
+  spotter_config.model_config.paraformer.decoder =
+      SHERPA_ONNX_OR(config->model_config.paraformer.decoder, "");
+
+  spotter_config.model_config.zipformer2_ctc.model =
+      SHERPA_ONNX_OR(config->model_config.zipformer2_ctc.model, "");
+
+  spotter_config.model_config.tokens =
+      SHERPA_ONNX_OR(config->model_config.tokens, "");
+  spotter_config.model_config.num_threads =
+      SHERPA_ONNX_OR(config->model_config.num_threads, 1);
+  spotter_config.model_config.provider =
+      SHERPA_ONNX_OR(config->model_config.provider, "cpu");
+  spotter_config.model_config.model_type =
+      SHERPA_ONNX_OR(config->model_config.model_type, "");
+  spotter_config.model_config.debug =
+      SHERPA_ONNX_OR(config->model_config.debug, 0);
+
+  spotter_config.max_active_paths =
+      SHERPA_ONNX_OR(config->max_active_paths, 4);
+
+  spotter_config.num_trailing_blanks =
+      SHERPA_ONNX_OR(config->num_trailing_blanks , 1);
+
+  spotter_config.keywords_score =
+      SHERPA_ONNX_OR(config->keywords_score, 1.0);
+
+  spotter_config.keywords_threshold =
+      SHERPA_ONNX_OR(config->keywords_threshold, 0.25);
+
+  spotter_config.keywords_file =
+      SHERPA_ONNX_OR(config->keywords_file, "");
+
+  if (config->model_config.debug) {
+    SHERPA_ONNX_LOGE("%s\n", spotter_config.ToString().c_str());
+  }
+
+  SherpaOnnxKeywordSpotter* spotter = new SherpaOnnxKeywordSpotter;
+
+  spotter->impl =
+    std::make_unique<sherpa_onnx::KeywordSpotter>(spotter_config);
+
+  return spotter;
+}
+
+void DestroyKeywordSpotter(SherpaOnnxKeywordSpotter* spotter) {
+  delete spotter;
+}
+
+SherpaOnnxOnlineStream* CreateKeywordStream(
+    const SherpaOnnxKeywordSpotter* spotter) {
+  SherpaOnnxOnlineStream* stream =
+    new SherpaOnnxOnlineStream(spotter->impl->CreateStream());
+  return stream;
+}
+
+int32_t IsKeywordStreamReady(
+    SherpaOnnxKeywordSpotter* spotter, SherpaOnnxOnlineStream* stream) {
+  return spotter->impl->IsReady(stream->impl.get());
+}
+
+void DecodeKeywordStream(SherpaOnnxKeywordSpotter* spotter,
+                         SherpaOnnxOnlineStream* stream) {
+  return spotter->impl->DecodeStream(stream->impl.get());
+}
+
+void DecodeMultipleKeywordStreams(
+    SherpaOnnxKeywordSpotter *spotter, SherpaOnnxOnlineStream **streams,
+    int32_t n) {
+  std::vector<sherpa_onnx::OnlineStream*> ss(n);
+  for(int32_t i=0; i!=n; ++i) {
```

Comment:
```suggestion
  for(int32_t i = 0; i != n; ++i) {
```

Please use `clang-format` to format your code.


You can install `clang-format` with
```
pip install clang-format
```